starknet_py/utils/typed_data.py

```diff
@@ -198,9 +238,9 @@ def _encode_value(
         ):
             encoded_value = self._encode_value_v0(basic_type, value)
         elif self.domain.resolved_revision == Revision.V1 and isinstance(
-            value, (str, int)
+            value, (str, int, dict)
```

Comment:
I don't think it can be omitted, because the range of types that `value` can be in `_encode_value_vX()` is smaller than in `_encode_value()`. Without this check, `value` could be for example of type `list`, which isn't considered in `_encode_value_v1()`.

Reply:
I overlooked the fact that this would cause some `pyright` errors, it can stay as it is.